case-lib/hijack.sh

```diff
@@ -48,9 +48,18 @@ function exit()
         done
     fi
 
-    # check if function already defined
-    # on exit check whether pulseaudio is disabled
-    [[ $(declare -f func_lib_restore_pulseaudio) ]] && func_lib_restore_pulseaudio
+    # check if function already defined.
+    # on exit check whether pulseaudio is disabled.
+    ret=0
+    if [[ $(declare -f func_lib_restore_pulseaudio) ]]; then
+        func_lib_restore_pulseaudio
+        ret=$?
+    fi
+    # if failed to restore pulseaudio, even test caes passed, set exit status to ret
```

Comment:
I did not get the logic here. can we make it to optional here? As we may have some device use **rename** method to disable pulseaudio. I think it would be better to allow pulseaudio failure as not all test case is depend on it.

But we do need to fail the test case if the test case need pulseaudio and we could not enable it.

Reply:
The code behaves like below:
- Cases don't touch PA, or use PA rename method: `[[ "${#PULSECMD_LST[@]}" -ne 0 ]] && return` and `[[ "${#PULSECMD_LST[@]}" -eq 0 ]] && return` will ensure `func_lib_disable_pulseaudio` and `func_lib_restore_pulseaudio` return immediately, `ret` will always be 0, and exit_status is the test case exit_status. Code change has no impact on this.
- Cases need to disable / enable PA: 
    - cases passed, PA disable/enable successfully, exit_status is test case exit status
    - cases failed, PA disable/enable status will be ignored, exit_status is test case exit status
    - cases passed, PA restore failed, exit_status will be replace with PA restore failed status
    - PA disable failed before running case, tag test case skipped(NA)

This is the logic, you can merge the PR into your local repo, and see it. I think there is no problem with the logic. BTW, PA in Ubuntu2004 will be managed by systemd, this allow us to disable PA easier.